gigascience/gigadb-website
# Only make the "Published" status selectable when a number of check boxes have been ticked

# Pull request for issue: #1684

This PR should be merged after https://github.com/gigascience/gigadb-website/pull/1730

This PR should be merged after https://github.com/gigascience/gigadb-website/pull/1699

This is a pull request for the following functionalities:

* Add 3 checkboxes that are required for the "Published" status to be enabled in the Upload Status dropbox
![image](https://github.com/gigascience/gigadb-website/assets/143437854/73ca8983-8914-4b39-86e9-60090cc0db28)


## How to test?

- Navigate to http://gigadb.gigasciencejournal.com/adminDataset/update/id/668
- Check that initially  upload status "published" is not selectable
- Check all the 3 checkboxes
- Verify that "published" is enabled
- optionally enable VoiceOver, navigate the form and confirm that both the title and description are read when focus enters the checkbox group
- Navigate to a published dataset, e.g. http://gigadb.gigasciencejournal.com/adminDataset/update/id/5, and verify that the checkboxes are not present

## How have functionalities been implemented?

* A new checkbox group has been added below the upload status dropdown.
* Javascript is used to programmatically toggle the disable attribute of the "published" option
* If all 3 checkboxes are checked, the "published" option is enabled, else, it's disabled
* A listener does the update any time any of the boxes is checked / unchecked
* The checkbox group title and description are announced by screen readers when the user focuses within
* If the upload status is "Published" when the page loads, then, the checkboxes are not shown as they are not needed

NOTE: the checkboxes are not programmatically checking whether that is true or false, the editor is trusted to be truthful when they check the checkboxes

## Any issues with implementation?

* Unclear what is the exact condition to programmatically determine whether a DOI exists

## Any changes to automated tests?

One acceptance test was failing because it tried to select an input by "DOI" label and thus it selected the newly added checkbox. By switching to use the input id instead, the issue is solved
